src/lib/operators/join_hash/join_hash_steps.hpp

```diff
@@ -231,7 +231,11 @@ std::vector<std::optional<HashTable<HashedType>>> build(const RadixContainer<Lef
         auto casted_value = type_cast<HashedType>(std::move(element.value));
         auto it = hashtable.find(casted_value);
         if (it != hashtable.end()) {
-          it->second.emplace_back(element.row_id);
+          if constexpr (!OneMatchPerValue) {
```

Comment:
"Match" is misleading here, there is no "match" yet, this is just the build phase and we're talking about hashtable entries, not (tuple-)"matches"